### PR TITLE
Fix error in graph standardization

### PR DESCRIPTION
**Pull request by AustinT**
The initial release included a file `syntheseus/search/graph/standardization.py` which contained code to turn all types of graphs into minimal AND/OR graphs (i.e. AND/OR graphs with one node per reaction and one node per molecule). 
It turns out that for MolSetGraph objects the code to fetch the reaction objects was incorrect and would crash. I didn't catch this before because I didn't actually write tests for this file.

This PR fixes this bug, and adds minimal "smoke" tests to make sure that this method runs without crashing. However, the actual output is not tested. To reflect this, the method still raises a warning when it is called.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -22,6 +22,7 @@ and the project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.
 
 ### Fixed
 
+- Fixed bug where standardizing MolSetGraphs crashed ([#24](https://github.com/microsoft/syntheseus/pull/24)) ([@austint])
```
Comment (kmaziarz) [inline CHANGELOG.md:25]: The [changelog format](https://common-changelog.org/#241-change) suggests using imperative mood for the entries (similarly to what's usually done for commit messages).
```suggestion
- Fix bug where standardizing MolSetGraphs crashed ([#24](https://github.com/microsoft/syntheseus/pull/24)) ([@austint])
```

File: syntheseus/tests/search/graph/test_standardization.py [hunk 2]
```diff
@@ -1 +1,27 @@
...
+        output = get_unique_node_andor_graph(andor_graph_non_minimal)
+
+    assert len(output) == len(andor_graph_non_minimal)  # no nodes deleted here
+
+
+def test_smoke_molset(
+    molset_tree_non_minimal: MolSetGraph,
+):
+    with pytest.warns(UserWarning):
+        get_unique_node_andor_graph(molset_tree_non_minimal)
```
Comment (kmaziarz) [inline syntheseus/tests/search/graph/test_standardization.py:27]: Just noting that these function signatures would normally fit on a single line, and the reason why `black` doesn't reformat this is due to the [magic-trailing comma behaviour](https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html#the-magic-trailing-comma) (it interprets the comma in the argument list as a strong preference against having the signature on a single line). I'm assuming it was not your intention to trigger this behaviour though 🙂
```suggestion
def test_smoke_andor(andor_graph_non_minimal: AndOrGraph):
    with pytest.warns(UserWarning):
        output = get_unique_node_andor_graph(andor_graph_non_minimal)

    assert len(output) == len(andor_graph_non_minimal)  # no nodes deleted here


def test_smoke_molset(molset_tree_non_minimal: MolSetGraph):
    with pytest.warns(UserWarning):
        get_unique_node_andor_graph(molset_tree_non_minimal)
```

**Review by kmaziarz — APPROVED** [review]
LGTM!

File: syntheseus/tests/search/graph/test_standardization.py (hunk 2 above)
Comment (AustinT) [reply syntheseus/tests/search/graph/test_standardization.py]: Good point, this was not intentional. I just fixed it.

**Pull request merged**